File: src/test/java/com/raychenon/leetcode/KthFactorTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 5/12/2020
 * https://leetcode.com/problems/the-kth-factor-of-n/
 */
public class KthFactorTest {

    private static KthFactor instance;

    @BeforeClass
    public static void setUp() {
        instance = new KthFactor();
    }


    @Test
    public void example1Test() {
        evaluateKthFactorMethods(3, 12, 3);
    }

    @Test
    public void example2Test() {
        evaluateKthFactorMethods(7, 7, 2);
    }

    @Test
    public void example3Test() {
        evaluateKthFactorMethods(-1, 4, 4);
    }

    void evaluateKthFactorMethods(int expected, int n, int k) {
        assertEquals(expected, instance.kthFactorLinear(n, k));
        assertEquals(expected, instance.kthFactorSqrt(n, k));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/KthFactorTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 5/12/2020
 * https://leetcode.com/problems/the-kth-factor-of-n/
 */
class KthFactorTest {

    @Test
    fun example1Test() {
        evaluateKthFactorMethods(3, 12, 3)
    }

    @Test
    fun example2Test() {
        evaluateKthFactorMethods(7, 7, 2)
    }

    @Test
    fun example3Test() {
        evaluateKthFactorMethods(-1, 4, 4)
    }

    fun evaluateKthFactorMethods(expected: Int, n: Int, k: Int) {
        TestCase.assertEquals(expected, KthFactor.kthFactorLinear(n, k))
        TestCase.assertEquals(expected, KthFactor.kthFactorSqrt(n, k))
    }
}